# ------------------------------------------------------------------------------------------------ #
# This patch fixes automatic aac and speex tagging in abcde-2.3.99.6
# and is an amalgamation of 2 separate patches from Ubuntu and Fedora:
# http://patches.ubuntu.com/a/abcde/abcde_2.3.99.6-1ubuntu2.patch
# http://cvs.fedoraproject.org/viewvc/rpms/abcde/F-10/abcde-2.3.99.6-speex-comment.patch?view=co
# ------------------------------------------------------------------------------------------------ #
diff -Naur abcde-2.3.99.6.orig/abcde abcde-2.3.99.6/abcde
--- abcde-2.3.99.6.orig/abcde	2006-08-06 05:14:00.000000000 +1000
+++ abcde-2.3.99.6/abcde	2009-03-17 21:07:11.000000000 +1100
@@ -1068,11 +1068,10 @@
 						*=*) ;;
 						*)   COMMENT="COMMENT=$COMMENT" ;;
 					esac	
-					COMMENT="--comment \"$COMMENT\""
 				fi
-				# Quick hack to avoid tagging Ogg/Speex, since there is no other way to tag than inline tagging
-				if [ ! "$DOTAG" = "y" ]; then
-					$RUN_COMMAND nice $EFFECTIVE_NICE $SPEEXENCODER $SPEEXENCODEROPTS --author "$TRACKARTIST" --title "$TRACKNAME" "$COMMENT" "$IN" "$OUT"
+				# Tag the file at encode time, as it can't be done after encoding.
+				if [ "$DOTAG" = "y" ]; then
+					$RUN_COMMAND nice $EFFECTIVE_NICE $SPEEXENCODER $SPEEXENCODEROPTS --author "$TRACKARTIST" --title "$TRACKNAME" ${COMMENT:+--comment "$COMMENT"} "$IN" "$OUT"
 				else
 					$RUN_COMMAND nice $EFFECTIVE_NICE $SPEEXENCODER $SPEEXENCODEROPTS "$IN" "$OUT"
 				fi
@@ -1087,8 +1086,8 @@
 				$RUN_COMMAND nice $EFFECTIVE_NICE $MPPENCODER $MPPENCODEROPTS --artist "$TRACKARTIST" --album "$DALBUM" --title "$TRACKNAME" --track "$1" --genre "$CDGENRE" --year "$CDYEAR" --comment "$COMMENT" "$IN" "$OUT"
 				;;
 			m4a)
-				# Quick hack to avoid tagging Ogg/Speex, since there is no other way to tag than inline tagging
-				if [ ! "$DOTAG" = "y" ]; then
+				# Tag the file at encode time, as it can't be done after encoding.
+				if [ "$DOTAG" = "y" ]; then
 					$RUN_COMMAND nice $EFFECTIVE_NICE $AACENCODER $AACENCODEROPTS --artist "$TRACKARTIST" --album "$DALBUM" --title "$TRACKNAME" --track "$1" --genre "$CDGENRE" --year "$CDYEAR" --comment "$COMMENT" -o "$OUT" "$IN"
 					
 				else
